components/style/properties/shorthand/position.mako.rs

```diff
@@ -138,6 +138,107 @@
 
 </%helpers:shorthand>
 
+% for kind in ["row", "column"]:
+<%helpers:shorthand name="grid-${kind}" sub_properties="grid-${kind}-start grid-${kind}-end"
+                    spec="https://drafts.csswg.org/css-grid/#propdef-grid-${kind}"
+                    products="gecko">
+    use values::specified::GridLine;
+    use parser::Parse;
+
+    pub fn parse_value(context: &ParserContext, input: &mut Parser) -> Result<Longhands, ()> {
+        let start = input.try(|i| GridLine::parse(context, i))?;
+        let end = if input.try(|i| i.expect_delim('/')).is_ok() {
+            GridLine::parse(context, input)?
+        } else {
+            let mut line = GridLine::default();
+            line.ident = start.ident.clone();       // ident from start value should be taken
+            line
+        };
+
+        Ok(Longhands {
+            grid_${kind}_start: start,
+            grid_${kind}_end: end,
+        })
+    }
+
+    impl<'a> ToCss for LonghandsToSerialize<'a> {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            self.grid_${kind}_start.to_css(dest)?;
+            let ref other = self.grid_${kind}_end;
+            if other.ident.is_some() || other.integer.is_some() {       // don't serialize auto
```

Comment:
It looks like `test / auto` and `auto / auto` serialize as are. Also `auto` serializes as `auto / auto` in Firefox and Chrome. So I guess we can skip this if and serialize it directly.